File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if root is None:
            return True
        if not self.isBalanced(root.right) or not self.isBalanced(root.left):
            return False       
        if abs(self.depth(root.right) - self.depth(root.left)) > 1:
            return False
        return True

    def depth(self, root: Optional[TreeNode]) -> bool:
        if root is None:
            return 0
        return 1 + max(self.depth(root.left), self.depth(root.right))
      

File: Data Structures & Algorithms/invert-a-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        Queue<TreeNode> myQ = new LinkedList<>();

        if (root != null)
            myQ.offer(root);
        else
            return root;

        while (myQ.size() > 0) {
            TreeNode currentNode = myQ.poll();
            TreeNode temp = currentNode.left;
            currentNode.left = currentNode.right;
            currentNode.right = temp;

            if (currentNode.left != null)
                myQ.offer(currentNode.left);
            if (currentNode.right != null)
                myQ.offer(currentNode.right);
        }

        return root;
    }
}
